Type definition might not be exported correctly

# What's happening

ref: https://github.com/fsubal/youdotfm/pull/22

Thank you for maintaining this project!

I tried adopting this polyfill in my TypeScript project, and turns out that importing as ponyfill results in a type error.

The following is the log for my Cloudflare pages (with Next.js 15.5.2 + TypeScript 5.2.2)

```
22:51:05.977 | > next build --turbopack
-- | --
22:51:05.978 |  
22:51:06.876 | ⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
22:51:06.881 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
22:51:06.881 | This information is used to shape Next.js' roadmap and prioritize features.
22:51:06.882 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
22:51:06.882 | https://nextjs.org/telemetry
22:51:06.882 |  
22:51:06.955 | ▲ Next.js 15.5.2 (Turbopack)
22:51:06.955 |  
22:51:07.065 | Creating an optimized production build ...
22:51:21.652 | ✓ Finished writing to disk in 437ms
22:51:21.693 | ✓ Compiled successfully in 13.9s
22:51:21.703 | Linting and checking validity of types ...
22:51:26.113 | Failed to compile.
22:51:26.114 |  
22:51:26.114 | ./src/app/sitemap.ts:1:26
22:51:26.114 | Type error: Cannot find module 'temporal-polyfill-lite' or its corresponding type declarations.
22:51:26.114 |  
22:51:26.115 | > 1 \| import { Temporal } from "temporal-polyfill-lite";
22:51:26.115 | \|                          ^
22:51:26.115 | 2 \| import { MetadataRoute } from "next";
22:51:26.115 | 3 \|
22:51:26.115 | 4 \| import { RelativeURL } from "../utils/url/internal";
22:51:26.160 | Next.js build worker exited with code: 1 and signal: null
22:51:26.200 | Failed: Error while executing user command. Exited with error code: 1
22:51:26.214 | Failed: build command exited with code: 1
22:51:27.236 | Failed: error occurred while running build command
```

I tried adding `declare module` in my d.ts and the type error is gone.

```ts
declare module "temporal-polyfill-lite" {
  export * from "temporal-polyfill-lite/dist/index";
}
```

# Expected behavior

`import { Temporal } from "temporal-polyfill-lite"` should resolve to the type in `temporal-polyfill-lite/dist/index.d.ts` .

--- a/README.md
+++ b/README.md
@@ -50,7 +50,7 @@ This polyfill doesn't internally rely on `bigint`, thus you can support older br
 
 This polyfill provides type definitions out of the box (You don't need `@types/*`).
 
-Note that this package uses the [`exports` field](https://nodejs.org/api/packages.html#exports) in our `package.json`, you need to set `"resolvePackageJsonExports": true`. Otherwise the type resolution will fail when you `import` this module.
+Note that this package uses the [`exports` field](https://nodejs.org/api/packages.html#exports) in our `package.json`, you need to set [`"resolvePackageJsonExports": true`](https://www.typescriptlang.org/tsconfig/#resolvePackageJsonExports) in your `tsconfig.json`. Otherwise the type resolution will fail when you `import` this module.
 
 ## Spec compliance
 

--- a/README.md
+++ b/README.md
@@ -50,7 +50,7 @@ This polyfill doesn't internally rely on `bigint`, thus you can support older br
 
 This polyfill provides type definitions out of the box (You don't need `@types/*`).
 
-Note that this package uses the [`exports` field](https://nodejs.org/api/packages.html#exports) in our `package.json`, you need to set `"resolvePackageJsonExports": true`.
+Note that this package uses the [`exports` field](https://nodejs.org/api/packages.html#exports) in our `package.json`, you need to set `"resolvePackageJsonExports": true`. Otherwise the type resolution will fail when you `import` this module.
 
 ## Spec compliance
 

--- a/README.md
+++ b/README.md
@@ -46,6 +46,12 @@ The polyfill works in browsers after September 2020 by default (e.g. Safari 14).
 
 This polyfill doesn't internally rely on `bigint`, thus you can support older browsers by transpiling and injecting polyfills (e.g. it relies on `globalThis` which is baseline since January 2020). However, if you use APIs accepting or returning `bigint`, `bigint` support is required.
 
+## TypeScript configuration
+
+This polyfill provides type definitions out of the box (You don't need `@types/*`).
+
+Note that this package uses the [`exports` field](https://nodejs.org/api/packages.html#exports) in our `package.json`, you need to set `"resolvePackageJsonExports": true`.
+
 ## Spec compliance
 
 It supports the latest spec with few intentional deviations (see `tests/expectedFailures` directory for details).

